ts/src/binance.ts

```diff
@@ -10385,8 +10385,14 @@ export default class binance extends Exchange {
         } else {
             throw new NotSupported (this.id + ' fetchLeverages() supports linear and inverse contracts only');
         }
+        let marketType = undefined;
+        if (symbols !== undefined) {
```

Comment:
I'm not sure if this really solves the issue because the user might not provide any symbol and we still have a conflicting ids scenario upon calling safeMarket

Reply:
you can test this out, the exception is not thrown. that's because, `safeMarket` gets marketType argument https://github.com/ccxt/ccxt/blob/master/ts/src/base/Exchange.ts#L4372

Reply:
in this case i think this is unnecessary because if needed, you can provide the already computed market type to parseLeverages  example: 
![image](https://github.com/ccxt/ccxt/assets/43336371/8923c4c3-3276-43ef-b465-959116d72c5e)

Reply:
that's why i provide 4th argument in parseLeverages. maybe you havent seen the last commit (which i pushed by the time i've commented).
as you decide, plz close this if unnecessary

Reply:
Yes this is not neccessary